libraries/Adapters/Microsoft.Bot.Builder.Adapters.WeChat/WeChatHttpAdapter.cs

```diff
@@ -0,0 +1,410 @@
+                        }
+                        else
+                        {
+                            responses[key] = new List<Activity> { activity };
+                        }
+
+                        break;
+                    default:
+                        _logger.LogInformation(
+                            $"WeChatAdapter.SendActivities(): Activities of type '{activity.Type}' aren't supported.");
+                        break;
+                }
+
+                resourceResponses.Add(new ResourceResponse(activity.Id));
+            }
+
+            return Task.FromResult(resourceResponses.ToArray());
+        }
+
+        /// <summary>
+        /// Process the request from WeChat.
+        /// </summary>
+        /// <param name="httpRequest">The request sent from WeChat.</param>
+        /// <param name="httpResponse">Http response object of current request.</param>
+        /// <param name="bot">The bot instance.</param>
+        /// <param name="secretInfo">Secret info for verify the request.</param>
+        /// <param name="passiveResponse">If using passvice response mode, if set to true, user can only get one reply.</param>
+        /// <param name="cancellationToken">Cancellation Token of this Task.</param>
+        /// <returns>Task running result.</returns>
+        public async Task ProcessAsync(HttpRequest httpRequest, HttpResponse httpResponse, IBot bot, SecretInfo secretInfo, bool passiveResponse, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            _logger.LogInformation("Receive a new request from WeChat.");
+            if (httpRequest == null)
+            {
+                throw new ArgumentNullException(nameof(httpRequest));
+            }
+
+            if (httpResponse == null)
+            {
+                throw new ArgumentNullException(nameof(httpResponse));
+            }
+
+            if (bot == null)
+            {
+                throw new ArgumentNullException(nameof(bot));
+            }
+
+            if (secretInfo == null)
+            {
+                throw new ArgumentNullException(nameof(secretInfo));
+            }
+
+            VerificationHelper.Check(secretInfo.Signature, secretInfo.Timestamp, secretInfo.Nonce, _token);
+
+            secretInfo.Token = _token;
+            secretInfo.EncodingAESKey = _encodingAESKey;
+            secretInfo.AppId = _appId;
+            var postDataDocument = XmlHelper.Convert(httpRequest.Body);
+            var wechatRequest = GetRequestMessage(postDataDocument, secretInfo);
+
+            try
+            {
+                // Reply WeChat(User) request have two ways, set response in http response or use background task to process the request async.
+                if (!passiveResponse)
+                {
+                    // Running a background task, Get bot response and parse it from activity to wechat response message
+                    if (_backgroundService == null)
```

Comment:
Move this validation to the constructor and ensure the parameter name is passed to ArgumentException.

Reply:
backgroud service is not required when using passive response mode.

Reply:
I see, let's leave this open for now. there may be a better way of handling this. 
Thanks